animation_nodes/algorithms/mesh_generation/find_shortest_path.pyx

```diff
@@ -0,0 +1,126 @@
+
+                for l in range(linkedVertices.length):
+                    linkedIndex = linkedVertices.data[l]
+                    weight = distanceVec3(vertices.data + currentIndex, vertices.data + linkedIndex) + weights.data[currentIndex]
+                    if weight < weights.data[linkedIndex]:
+                        weights.data[linkedIndex] = weight
+                        previousVertices.data[linkedIndex] = currentIndex
+
+    cdef LongList indices
+    cdef Py_ssize_t index, amount
+    cdef Vector3DList sortLocations
+    cdef list meshes, splines, strokes
+
+    if mode == "PATH":
+        indices = LongList()
+        index = targets.data[0]
+        if previousVertices.data[index] == -1: return indices
+        for j in range(vertexCount):
+            indices.append(index)
+            index = previousVertices.data[index]
+            if index == -1: break
+
+        return indices.reversed()
+
+    if pathType == "MESH":
+        meshes = []
+        for i in range(vertexCount):
+            sortLocations = Vector3DList()
+            index = i
+            if previousVertices.data[index] == -1: continue
+            for j in range(vertexCount):
+                sortLocations.append(vertices[index])
+                index = previousVertices.data[index]
+                if index == -1: break
+
+            meshes.append(getLinesMesh(sortLocations.reversed(), False))
+
+        return meshes
+
+    elif pathType == "SPLINE":
+        splines = []
+        for i in range(vertexCount):
+            sortLocations = Vector3DList()
+            index = i
+            if previousVertices.data[index] == -1: continue
+            for j in range(vertexCount):
+                sortLocations.append(vertices[index])
+                index = previousVertices.data[index]
+                if index == -1: break
+
+            splines.append(PolySpline.__new__(PolySpline, sortLocations.reversed()))
+
+        return splines
+
+    elif pathType == "STROKE":
+        strokes = []
+        for i in range(vertexCount):
+            sortLocations = Vector3DList()
+            index = i
+            if previousVertices.data[index] == -1: continue
+            for j in range(vertexCount):
+                sortLocations.append(vertices[index])
+                index = previousVertices.data[index]
+                if index == -1: break
+
+            amount = sortLocations.length
+            strengths = FloatList(length = amount)
+            pressures = FloatList(length = amount)
+            uvRotations = FloatList(length = amount)
+            vertexColors = ColorList(length = amount)
+            strengths.fill(1)
+            pressures.fill(1)
+            uvRotations.fill(0)
+            vertexColors.fill((0, 0, 0, 0))
+            strokes.append(GPStroke(sortLocations.reversed(), strengths, pressures,
+                                    uvRotations, vertexColors, 10))
```

Comment:
Seems like all of those share the same code. You should deduplicate.

Reply:
Done.